src/algorithm.rs

```diff
@@ -85,30 +85,58 @@ where
 /// Naive version of the algorithm that given a full Network and a precomputed
 /// set W of random walks, iterates over each edge of the Network and computes
 /// the osrank.
-pub fn osrank_naive<L, G>(
+pub fn osrank_naive<L, G, RNG>(
     seed_set: Option<SeedSet>,
     network: &mut G,
     ledger_view: &L,
     iter: i32,
+    initial_seed: <RNG as SeedableRng>::Seed,
     get_weight: Box<Fn(&<G::Edge as GraphObject>::Metadata) -> f64>,
     from_osrank: Box<(Fn(&G::Node, Osrank) -> Metadata<G::Node>)>,
 ) -> Result<(), OsrankError>
 where
     L: LedgerView,
     G: Graph,
     Id<G::Node>: Clone + PartialEq,
+    RNG: Rng + SeedableRng,
+    <RNG as SeedableRng>::Seed: Clone,
 {
+    //NOTE(adn) The fact we are creating a new RNG every time we call
+    // `random_walk` is deliberate and something to think about. We probably
+    // want to "restart" the randomness from the initial seed every call to
+    // `random_walk`, which means this function has to consume the RNG.
```

Comment:
If we think that's OK to have the RNG "resume" from where the previous walk left, we can avoid cloning & reconstructing it every time.

Reply:
Do you have any arguments against that in mind?

Reply:
I don't -- I just assumed we wanted to reset everything to a clean slate before proceeding. Either way, it's a trivial change, so we can defer it for later.